flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/FlinkCalciteCatalogReader.java

```diff
@@ -104,6 +115,38 @@ public Prepare.PreparingTable getTable(List<String> names) {
         }
     }
 
+    public @Nullable CatalogSchemaModel getModel(List<String> names) {
+        for (List<String> schemaPath : getSchemaPaths()) {
+            CalciteSchema schema =
+                    getSchema(
+                            getRootSchema(),
+                            Iterables.concat(schemaPath, Util.skipLast(names)),
+                            nameMatcher());
+            if (schema == null) {
+                continue;
+            }
+
+            if (!(schema.schema instanceof FlinkSchema)) {
+                throw new ValidationException("getModel() only supports FlinkSchema.");
+            }
+
+            FlinkSchema flinkSchema = (FlinkSchema) schema.schema;
+            String modelName =
+                    nameMatcher().isCaseSensitive()
+                            ? Util.last(names)
+                            : caseInsensitiveLookup(flinkSchema.getModelNames(), Util.last(names));
+            if (modelName == null) {
+                return null;
```

Comment:
why to we return null instead of continue? Can we not have a schemaPath that we hit later that might have a valid model? I suspect the proposal from @fsk119 will avoid this.

Reply:
+1. We should continue to iterate other schemas.